server/src/main/java/io/crate/execution/ddl/tables/MappingUtil.java

```diff
@@ -40,21 +43,27 @@
 
 import static io.crate.metadata.Reference.buildTree;
 import static io.crate.metadata.table.ColumnPolicies.ES_MAPPING_NAME;
+import static org.elasticsearch.cluster.metadata.Metadata.COLUMN_OID_UNASSIGNED;
 
 public class MappingUtil {
 
     /**
      * This is a singe entry point to creating mapping: adding a column(s), create a table, create a partitioned table (template).
      * @param tableColumnPolicy has default value STRICT if not specified on a table creation.
      * On column addition it's NULL in order to not override an existing value.
+     *
+     * @param currentOID is the starting point for the further assignment. Must be -1 if used in BWC code which doesn't expect oid in the mapping.
+     * @param metadataBuilder is mutated to set updated OID on the callers side. Must be NULL if used in BWC code which doesn't expect oid in the mapping.
      */
     public static Map<String, Object> createMapping(List<Reference> columns,
                                                     IntArrayList pKeyIndices,
                                                     Map<String, String> checkConstraints,
                                                     Map<String, Object> indices,
                                                     List<List<String>> partitionedBy,
                                                     @Nullable ColumnPolicy tableColumnPolicy,
-                                                    @Nullable String routingColumn) {
+                                                    @Nullable String routingColumn,
+                                                    long currentOID,
```

Comment:
```suggestion
                                                    long maxOID,
```

For consistency with our terminology around seqNo/checkpoints